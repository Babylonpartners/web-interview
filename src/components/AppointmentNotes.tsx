import * as React from 'react'
import { headerStyles } from '../styles/styles';

interface Props {
    handleChange: any
    appointmentNotesError: any
}

export class AppointmentNotes extends React.PureComponent<Props> {
    render() {
        return (

        <div data-testid="appointmentNotesComponent" >
            <strong
                style={headerStyles}
                data-testid="appointmentNotesHeader" 
            >
                Notes
            </strong>
            <br />
            <textarea
              style={headerStyles}
              data-testid="appointmentNotesTextArea" 
              className="notes"
              placeholder="Describe your symptoms"
              onChange={this.props.handleChange}
            />
            <br />
            <br />
            <br />
            <br />
            <br />
          </div>

        );
}}