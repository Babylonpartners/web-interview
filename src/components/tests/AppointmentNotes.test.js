import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { AppointmentNotes } from '../AppointmentNotes'

afterEach(cleanup)

test('AppointmentNotes>', () => {
  const spy = jest.fn()
  const { queryByTestId, container } = render(
    <AppointmentNotes handleChange={spy} appointmentNotesError />
  )
  expect(container.firstChild).toMatchSnapshot()
  expect(queryByTestId('appointmentNotesHeader')).toBeTruthy()
  expect(queryByTestId('appointmentNotesTextArea')).toBeTruthy()
})
