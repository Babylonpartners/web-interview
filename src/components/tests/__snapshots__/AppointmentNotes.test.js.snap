// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentNotes> 1`] = `
<div
  data-testid="appointmentNotesComponent"
>
  <strong
    data-testid="appointmentNotesHeader"
    style="color: rgb(72, 72, 72); font-family: 'Roboto', sans-serif; margin-bottom: 5px;"
  >
    Notes
  </strong>
  <br />
  <textarea
    class="notes"
    data-testid="appointmentNotesTextArea"
    placeholder="Describe your symptoms"
    style="color: rgb(72, 72, 72); font-family: 'Roboto', sans-serif; margin-bottom: 5px;"
  />
  <br />
  <br />
  <br />
  <br />
  <br />
</div>
`;
